Magento_Theme/layout/default.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © 2015 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
+    <body>
+
+        <move element="header.panel" destination="header.wrapper"/>
+        <move element="logo" destination="header.wrapper" before="-" />
+        <move element="top.links" destination="header.wrapper" after="logo" />
+        <move element="catalog.topnav" destination="header.wrapper" after="top.links" />
+
+        <referenceContainer name="header-wrapper" label="Page Header" as="header-wrapper" htmlTag="div" htmlClass="header-content"/>
+
+        <container name="header-wrapper" label="Page Header" as="header-wrapper" htmlTag="div" htmlClass="header content">
+            <block class="Magento\Theme\Block\Html\Header\Logo" name="logo">
+                <arguments>
+                    <argument name="logo_file" xsi:type="string">images/logo.png</argument>
```

Comment:
As discussed, that will be installed as database content and then transferred to the file system through the admin UI.  So the logo file doesn't have to be part of the theme.